keyvault/client/src/main/java/com/azure/keyvault/SecretAsyncClientBuilder.java

```diff
@@ -0,0 +1,230 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * Configures and builds instance of {@link SecretAsyncClient client} to manage {@link Secret secrets} in the specified key vault.
+ *
+ * <p> A {@link Secret secret} is a resource managed by Key Vault. It is represented by non-null fields secret.name and secret.value.
+ * The secret.expires, secret.contentType and secret.notBefore values in {@code secret} are optional. The secret.enabled
+ * field is set to true by Azure Key Vault, if not specified. The secret.id, secret.created, secret.updated, secret.recoveryLevel
+ * fields are auto assigned when the secret is created in the key vault. </p>
+ *
+ * <p> Minimal configuration options required by {@link SecretAsyncClientBuilder secretClientBuilder} to build {@link SecretAsyncClient}
+ * are {@link String vaultEndPoint} and {@link AsyncServiceClientCredentials credentials}. If a custom {@link HttpPipeline pipeline}
+ * is specified as configuration option, then no other configuration option needs to be specified. </p>
+ *
+ * <pre>
+ *    SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+ *                                .vaultEndPoint("https://myvault.vault.azure.net/")
+ *                                .credentials(keyVaultCredentials)
+ *                                .build()
+ *
+ *    SecretAsyncClient secretAsyncClientWithCustomPipeline = SecretAsyncClient.builder()
+ *                                                  .pipeline(customHttpPipeline)
+ *                                                  .build()
+ * </pre>
+ *
+ * <p> The {@link HttpLogDetailLevel logdetailLevel}, multiple custom {@link HttpPipeline policies} and custom
+ * {@link HttpClient httpClient} be optionally configured in the {@link SecretAsyncClientBuilder}.</p>
+ *
+ * <pre>
+ *    SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+ *                                .vaultEndPoint("https://myvault.vault.azure.net/")
+ *                                .credentials(keyVaultCredentials)
+ *                                .httpLogDetailLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+ *                                .addPolicy(customPolicyOne)
+ *                                .addPolicy(customPolicyTwo)
+ *                                .httpClient(client)
+ *                                .build()
+ * </pre>
+ */
+public final class SecretAsyncClientBuilder {
+    private final List<HttpPipelinePolicy> policies;
+    private AsyncServiceClientCredentials credentials;
+    private HttpPipeline pipeline;
+    private URL vaultEndPoint;
+    private HttpClient httpClient;
+    private HttpLogDetailLevel httpLogDetailLevel;
+    private RetryPolicy retryPolicy;
+    private String userAgent;
+
+    SecretAsyncClientBuilder() {
+        userAgent = String.format("Azure-SDK-For-Java/%s (%s)", AzureKeyVaultConfiguration.SDK_NAME, AzureKeyVaultConfiguration.SDK_VERSION);
```

Comment:
I've merged in a new user agent policy whose constructor takes the SDK name and version so you can use that.

Reply:
Updated.